pkg/master/master.go

```diff
@@ -343,19 +342,19 @@ type restOptionsFactory struct {
 	storageDecorator        generic.StorageDecorator
 }
 
-func (f restOptionsFactory) NewFor(resource schema.GroupResource) generic.RESTOptions {
+func (f *restOptionsFactory) GetRESTOptions(resource schema.GroupResource) (*generic.RESTOptions, error) {
 	storageConfig, err := f.storageFactory.NewConfig(resource)
 	if err != nil {
-		glog.Fatalf("Unable to find storage destination for %v, due to %v", resource, err.Error())
+		return nil, fmt.Errorf("Unable to find storage destination for %v, due to %v", resource, err.Error())
```

Comment:
Actually, not returning error here but `glog.Fatalf` would drop all those panics from `*/etcd.go` files?